lib/collector_registration.rb

```diff
@@ -0,0 +1,139 @@
+require 'uri'
+
+require 'global_configuration'
+require 'json'
+require 'logging'
+require 'rest_client_extensions'
+
+class CollectorRegistration
+  include GlobalConfiguration
+  include Logging
+
+  def initialize
+    @environment = ENV['METER_ENV'] || 'development'
+    @configuration = GlobalConfiguration::GlobalConfig.instance
+  end
+
+  def configure_uc6
+    if credentials_present?
+      verify_connection_from_credentials
+    else #no credentials
+      if !@configuration.blank_value?(:uc6_api_host)
+        verify_connection_retrieving_missing_info
+      else
+        logger.error "Pending configuration of credentials"
+        exit(1)
+      end
+    end
+  end
+
+  private
+
+  def credentials_present?
+    !@configuration.blank_value?(:uc6_application_id) && !@configuration.blank_value?(:uc6_application_secret) &&
```

Comment:
maybe we could create another method called present_value? which will perform the same !...blank_value? so that we could make the code easier to read